Hide bottom tab completed

AliNisarAhmed opened the pull request:
## Description

This PR will 
- create AppRoutes enum for typechecked routes throughout the apps
- Hide bottom tabs from completed screen

## Ticket Link

  Closes https://github.com/heylinda/heylinda-app/issues/37

## How has this been tested?

Tested the Completed screen using these devices:

Pixel 4 XL API 30 - Simulator

## Screenshots

(To be added)

## Checklist

- [x] Added a "Closes [issue number]" in the ticket link section
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens

AliNisarAhmed commented:
@watadarkstar Please take a look and let me know what you think about the `AppRoutes.ts` file which contains an enum describing all the routes and screens in the App. 

To hide the bottom tabs, I added an intermediate Stack Navigator, called `MainNavigator`, between `Root` and `BottomTabsNavigator`.

watadarkstar commented:
Hey @AliNisarAhmed the underlying solution makes sense to me! Nice work!

However, I'm not 100% sure I love the new enum as we already have TypeScript checking the naming of things. It also creates another layer of abstraction and I wonder if its worth it / solves a problem. What benefit do you see in using an enum? I'll give it some more thought as I'm not 100% convinced.

AliNisarAhmed commented:
@watadarkstar I added `AppRoutes` enum because line #23 in `CompletedScreen/index.tsx` would not type check, but that was a different problem which was solved by using `CompositeNavigationProp` in PlayScreen.

Though AppRoutes does provide a unified "interface" to all routes/screens in the app, I think you are right. I checked all the route/screen strings in the app, and they all type check, and there is no possibility of using a wrong string. So we do not need the `AppRoutes` enum which is just adds overhead for little benefit.

I have removed it from the code, going to push it soon. After I upload a screenshot, PR should be ready for final review.

Thanks for the feedback.

watadarkstar commented on screens/Completed/index.tsx:
```diff
@@ -6,21 +6,21 @@ import { Button } from 'react-native-paper'
     Linking.openURL('https://opencollective.com/heylinda/donate')
   }
-  const onPressSkip = () => navigation.navigate('HomeScreen')
+  const onPressSkip = () => navigation.replace('Main')
```
I had to make a small change here from `navigate` to `replace`

watadarkstar reviewed (approved):
LGTM - nice work! 👍  I made a small change to the PR just to get it merged in faster! Hope thats cool!
